core/templates/services/playthrough.service.spec.ts

```diff
@@ -144,7 +144,7 @@ describe('PlaythroughService', () => {
         playthroughService.recordPlaythrough(false);
 
         let postData = {
-          playthrough_data: playthroughService.playthrough.toBackendDict(),
+          playthrough_data: PlaythroughService.playthrough.toBackendDict(),
```

Comment:
Hi @srijanreddy98, I have updated the class variables to static in `PlaythroughService` as well which I upgraded earlier. I am facing an issue after that. The four tests in this file which are changed are failing when run together but they pass when run individually -- is there any other change required here? Thanks!

Reply:
Indeed, tests are failing. I reset the last three commits in this pr and tried changing the variables to static myself. After doing that,  two tests still fail.
https://github.com/ankita240796/oppia/compare/upgrade-3...srijanreddy98:up-3?expand=1
I am looking at solving the two tests that are failing, maybe the other two popped up due to the other commits (doesn't look like it but still better to take a look).

> The four tests in this file which are changed are failing when to run together but they pass when run individually

This is weird, there are seven tests in the `'recording exploration playthroughs'` describe block and out of the seven all of the four tests (including the two tests failing in my branch) that are failing come in the `.recordPlaythrough()` describe block.

I don't have a real answer to this, I am trying to debug the test, will let you know if I find something.

@bansalnitish, can you think of anything that could have gone wrong here.

Reply:
Thanks @srijanreddy98, I will try to look into it as well.

Reply:
This is fixed now - in the function `initSession`, we need to initialise all the static vars which wasn't being done before.

Reply:
Oh, that's why they were succeeding individually...